virtcontainers/types/container_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func testContainerStateTransition(t *testing.T, state StateString, newState StateString) error {
+	s := ContainerState{
+		State: state,
+	}
+
+	return s.ValidTransition(state, newState)
+}
+
+func TestContainerStateReadyRunning(t *testing.T) {
+	err := testContainerStateTransition(t, StateReady, StateRunning)
+	assert.NoError(t, err)
+}
+
+func TestContainerStateRunningPaused(t *testing.T) {
+	err := testContainerStateTransition(t, StateRunning, StatePaused)
+	assert.NoError(t, err)
+}
+
+func TestContainerStatePausedRunning(t *testing.T) {
+	err := testContainerStateTransition(t, StatePaused, StateRunning)
+	assert.NoError(t, err)
+}
+
+func TestContainerStatePausedStopped(t *testing.T) {
+	err := testContainerStateTransition(t, StatePaused, StateStopped)
+	assert.NoError(t, err)
+}
+
+func TestContainerStateRunningStopped(t *testing.T) {
+	err := testContainerStateTransition(t, StateRunning, StateStopped)
+	assert.NoError(t, err)
+}
+
+func TestContainerStateReadyStopped(t *testing.T) {
+	err := testContainerStateTransition(t, StateReady, StateStopped)
+	assert.NoError(t, err)
+}
+
+func TestContainerStateStoppedRunning(t *testing.T) {
+	err := testContainerStateTransition(t, StateStopped, StateRunning)
+	assert.NoError(t, err)
+}
+
+func TestContainerStateStoppedReady(t *testing.T) {
+	err := testContainerStateTransition(t, StateStopped, StateReady)
+	assert.Error(t, err)
+}
+
+func TestContainerStatePausedPaused(t *testing.T) {
+	err := testContainerStateTransition(t, StatePaused, StatePaused)
+	assert.Error(t, err)
+}
+
+func testContainerStateValid(t *testing.T, stateStr StateString, expected bool) {
+	state := &ContainerState{
+		State: stateStr,
+	}
+
+	ok := state.Valid()
+	assert.Equal(t, ok, expected)
```

Comment:
I'd consider adding a string at the end of this call containing `state` and `expected`) to aid with debugging. It's a good habit to get into, particularly when trying to debug "remote failures" (those annoying tests that work locally but fail in the CI ;-)

Reply:
@jodh-intel thanks for the comment, I added a message here